netbox/dcim/models/devices.py

```diff
@@ -1062,3 +1063,83 @@ def delete(self, *args, **kwargs):
             )
 
         return super().delete(*args, **kwargs)
+
+
+class VirtualDeviceContext(PrimaryModel):
+    device = models.ForeignKey(
+        to='Device',
+        on_delete=models.PROTECT,
+        related_name='vdcs',
+        blank=True,
+        null=True
+    )
+    name = models.CharField(
+        max_length=64
+    )
+    status = models.CharField(
+        max_length=50,
+        blank=True,
+        choices=VirtualDeviceContextStatusChoices,
+    )
+    identifier = models.PositiveSmallIntegerField(
```

Comment:
Should probably add a `help_text` here to explain the field's intended purpose.

Reply:
> Unique identifier provided by the platform being virtualized (Example: Nexus VDC Identifier)